### ISSUE TITLE
Cannot parse network join response

### ISSUE THREAD

**Issue by pawelka** [doudz/zigate#40]
Zigate firmware 3.0e
Zigate library 0.21.1 

I cannot start network, but when I'm using TestGUI for ZiGate network starts.

Zigate lib logs:

>>> z = zigate.connect(port=None) # Leave None to auto-discover the port
DEBUG:zigate:Try loading persistent file
DEBUG:zigate:Load success
INFO:zigate:Searching ZiGate port
INFO:zigate:ZiGate found at /dev/cu.usbserial
DEBUG:zigate:REQUEST : 0x0010
DEBUG:zigate:Msg to send b'0010000010'
DEBUG:zigate:Encoded Msg to send b'\x01\x02\x10\x10\x02\x10\x02\x10\x10\x03'
DEBUG:zigate:Waiting for status message for command 0x0010
DEBUG:zigate:Received response 0x8000: b'\x00\x00\x00\x10'
DEBUG:zigate:RESPONSE 0x8000 - Status response : status:0, sequence:0, packet_type:16, error:b'', rssi:0
DEBUG:zigate:Dispatch ZIGATE_RESPONSE_RECEIVED
DEBUG:zigate:Received response 0x8010: b'\x00\x02\x03\x0e'
DEBUG:zigate:RESPONSE 0x8010 - Version list : major:2, installer:30e, rssi:0, version:3.0e
DEBUG:zigate:Dispatch ZIGATE_RESPONSE_RECEIVED
DEBUG:zigate:STATUS code to command 0x0010:0
DEBUG:zigate:Waiting for message 0x8010
DEBUG:zigate:Stop waiting, got message 0x8010
DEBUG:zigate:REQUEST : 0x0021 b'\x03\x18\xc8\x00'
DEBUG:zigate:Msg to send b'00210004f60318c800'
DEBUG:zigate:Encoded Msg to send b'\x01\x02\x10!\x02\x10\x02\x14\xf6\x02\x13\x18\xc8\x02\x10\x03'
DEBUG:zigate:Waiting for status message for command 0x0021
ERROR:zigate:OOPS connection lost, reconnect...
DEBUG:zigate:Received response 0x8000: b'\x00\x00\x00!'
DEBUG:zigate:RESPONSE 0x8000 - Status response : status:0, sequence:0, packet_type:33, error:b'', rssi:0
DEBUG:zigate:Dispatch ZIGATE_RESPONSE_RECEIVED
DEBUG:zigate:STATUS code to command 0x0021:0
DEBUG:zigate:REQUEST : 0x0023 b'\x00'
DEBUG:zigate:Msg to send b'002300012200'
DEBUG:zigate:Encoded Msg to send b'\x01\x02\x10#\x02\x10\x02\x11"\x02\x10\x03'
DEBUG:zigate:Waiting for status message for command 0x0023
DEBUG:zigate:Received response 0x8000: b'\x00\x00\x00#'
DEBUG:zigate:RESPONSE 0x8000 - Status response : status:0, sequence:0, packet_type:35, error:b'', rssi:0
DEBUG:zigate:Dispatch ZIGATE_RESPONSE_RECEIVED
DEBUG:zigate:STATUS code to command 0x0023:0
DEBUG:zigate:Check network state
DEBUG:zigate:REQUEST : 0x0024
DEBUG:zigate:Msg to send b'0024000024'
DEBUG:zigate:Encoded Msg to send b'\x01\x02\x10$\x02\x10\x02\x10$\x03'
DEBUG:zigate:Waiting for status message for command 0x0024
DEBUG:zigate:Received response 0x8024: b'\x04'
Exception in thread Thread-46:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/pkarpins/projects/home/home-assistant/lib/python3.7/site-packages/zigate/core.py", line 155, in _event_loop
    self.decode_data(packet)
  File "/Users/pkarpins/projects/home/home-assistant/lib/python3.7/site-packages/zigate/core.py", line 373, in decode_data
    response = RESPONSES.get(msg_type, Response)(value, rssi)
  File "/Users/pkarpins/projects/home/home-assistant/lib/python3.7/site-packages/zigate/responses.py", line 46, in __init__
    self.decode()
  File "/Users/pkarpins/projects/home/home-assistant/lib/python3.7/site-packages/zigate/responses.py", line 107, in decode
    sdata, msg_data = self.__decode(fmt, keys, msg_data)
  File "/Users/pkarpins/projects/home/home-assistant/lib/python3.7/site-packages/zigate/responses.py", line 118, in __decode
    sdata = OrderedDict(zip(keys, struct.unpack(fmt, data[:size])))
struct.error: unpack requires a buffer of 12 bytes

WARNING:zigate:No response after command 0x0024
DEBUG:zigate:REQUEST : 0x0009
DEBUG:zigate:Msg to send b'0009000009'
DEBUG:zigate:Encoded Msg to send b'\x01\x02\x10\x02\x19\x02\x10\x02\x10\x02\x19\x03'
DEBUG:zigate:Waiting for status message for command 0x0009

TestGUI Logs:
00:29:06.436 -> 01 02 10 10 02 10 02 10 10 03 
00:29:06.483 <- 01 80 00 00 05 95 00 00 00 10 00 03
00:29:06.545 <- 01 80 10 00 05 9A 00 02 03 0E 00 03
00:29:07.843 -> 01 02 10 21 02 10 02 14 2D 02 10 02 10 02 18 02 10 03 
00:29:07.920 <- 01 80 00 00 05 A4 00 00 00 21 00 03
00:29:16.952 -> 01 02 10 23 02 10 02 11 22 02 10 03 
00:29:17.030 <- 01 80 00 00 05 A6 00 00 00 23 00 03
00:29:19.030 -> 01 02 10 24 02 10 02 10 24 03 
00:29:19.123 <- 01 80 24 00 02 A2 04 00 03
00:29:19.186 <- 01 80 00 00 05 A1 00 00 00 24 00 03

**Comment by pawelka** [comment]
If I understand zigate commands correctly this is because command was send too fast. TestGUI works because of manual delays.

"4 = Busy (Node is carrying out a Lengthy operation and is Currently Unable to handle the incoming command)" - it's a description for 8000, but there is a chance that applies for network join too.

**Comment by doudz** [comment]
Yes you're that's the same status code
I have to handle this case
is it happening everytimes ?

**Comment by pawelka** [comment]
Everytime :-/

I added (after learing python basics) 3s delay in your lib before start network and this change didn’t help :-/

**Comment by doudz** [comment]
Did you add time.sleep(3) just before start_network() at line 248 ?
```
        self.set_type(TYPE_COORDINATOR)
        LOGGER.debug('Check network state')
        time.sleep(3)
        self.start_network()
        network_state = self.get_network_state()
```

**Comment by pawelka** [comment]
Yes, this was my change.

**Comment by doudz** [comment]
Ok I'll handle it
I think it's related to https://github.com/fairecasoimeme/ZiGate/issues/15

I think there's no problem, the busy status probably just indicate that the network is already started.
In previous firmware, this command just not answer.

**Comment by doudz** [comment]
Should be fixed in 0.21.2

**Comment by pawelka** [comment]
Works, thank you.
